src/ucc/api/ucc.h

```diff
@@ -1574,5 +1588,229 @@ static inline ucc_status_t ucc_collective_test(ucc_coll_req_h request)
  */
 ucc_status_t ucc_collective_finalize(ucc_coll_req_h request);
 
+
+/**
+ * @ingroup UCC_EVENT_DT
+ *
+ */
+
+typedef enum ucc_event_type {
+	UCC_EVENT_COLLECTIVE_POST       = UCC_BIT(0),
+	UCC_EVENT_COLLECTIVE_COMPLETE   = UCC_BIT(1),
+	UCC_EVENT_COMPUTE_COMPLETE      = UCC_BIT(2),
+	UCC_EVENT_OVERFLOW              = UCC_BIT(3)
+} ucc_event_type_t;
+
+/**
+ * @ingroup UCC_EVENT_DT
+ *
+ */
+typedef enum ucc_ee_type {
+    UCC_CUDA_STREAM = UCC_BIT(0),
+	UCC_CPU_THREAD  = UCC_BIT(1),
+	UCC_DPU_THREAD  = UCC_BIT(2)
+} ucc_ee_type_t;
+
+/**
+ * @ingroup UCC_EVENT_DT
+ *
+ */
+typedef struct ucc_event_context {
+    ucc_event_type_t     ev_type;
+    size_t 		         info_size;
+    void 		         *info;
+} ucc_event_context_t;
+
+/**
+ * @ingroup UCC_EVENT_DT
+ *
+ */
+typedef struct ucc_event {
+	ucc_event_type_t        ev_type;
+	ucc_event_context_t     *event_context;
+	ucc_coll_req_h          req;
+} ucc_ev_t;
+
+/**
+ * @ingroup UCC_EVENT_DT
+ *
+ */
+typedef enum ucc_event_queue_type {
+	UCC_POST_QUEUE          = UCC_BIT(0),
+	UCC_COMPLETION_QUEUE    = UCC_BIT(1)
+} ucc_event_queue_type_t;
+
+/**
+ * @ingroup UCC_EVENT_DT
+ *
+ */
+typedef struct ucc_ee_params {
+     ucc_ee_type_t 			        ee_type;
+     ucc_event_queue_type_t        	ee_eq;
```

Comment:
what does ee_queue type in ee? shouldn't ee have both post and complete queue? do we really need two queues?